crates/transaction-pool/src/maintain.rs

```diff
@@ -494,6 +495,82 @@ pub async fn maintain_transaction_pool<N, Client, P, St, Tasks>(
 
                 // keep track of mined blob transactions
                 blob_store_tracker.add_new_chain_blocks(&blocks);
+
+                // At Osaka transition, we need to convert blobs to new format
+                if !chain_spec.is_osaka_active_at_timestamp(tip.timestamp()) &&
+                    chain_spec.is_osaka_active_at_timestamp(tip.timestamp().saturating_add(12))
```

Comment:
it's technically possible that we reach this twice if we go through a reorg but imo we can ignore this case